Character.cpp

```diff
@@ -98,20 +98,43 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        if (hitPoints < *initialHitPoints)
+        {
+            hitPoints = *initialHitPoints;
+        }        
+        
+        hitPoints *= 1.1;
+        *initialHitPoints = hitPoints;
+        
+        if (attackDamage < *initialAttackDamage)
+        {
+            attackDamage = *initialAttackDamage;
+        }
+        
+        attackDamage *= 1.1;
+        *initialAttackDamage = attackDamage;
+
+        if (armor < *initialArmorLevel)
+        {
+            armor = *initialArmorLevel;
+        }
+        armor *= 1.1;
+        *initialArmorLevel = armor;
```

Comment:
You are doing the same thing three times with different pairs of variables.
Recognize the pattern.
Don't repeat yourself.
Write a free function and call it three times.  The argument list should not include `std::unique_ptr`.
- [ ] revise